contracts/contracts/staking/SingleAssetStaking.sol

```diff
@@ -377,14 +388,21 @@ contract SingleAssetStaking is Initializable, Governable {
      * @param duration Number of seconds this stake will be held for
      */
     function stake(uint256 amount, uint256 duration) external requireLiquidity {
-        require(amount > 0, "Cannot stake 0");
-
-        uint240 rewardRate = _findDurationRate(duration);
-        require(rewardRate > 0, "Invalid duration"); // we couldn't find the rate that correspond to the passed duration
+        // no checks are performed in this function since those are already present in _stakeWithChecks
+        _stakeWithChecks(msg.sender, amount, duration);
+    }
 
-        _stake(msg.sender, USER_STAKE_TYPE, duration, rewardRate, amount);
-        // transfer in the token so that we can stake the correct amount
-        stakingToken.safeTransferFrom(msg.sender, address(this), amount);
+    /**
+     * @dev Stake an approved amount of staking token into the contract. This function
+     *      can only be called by OGN token contract.
+     * @param staker Address of the account that is creating the stake
+     * @param amount Number of tokens to stake in 1e18
+     * @param duration Number of seconds this stake will be held for
+     */
+    function stakeWithSender(address staker, uint256 amount, uint256 duration) external returns (bool) {
+        // no checks are performed in this function since those are already present in _stakeWithChecks
```

Comment:
Important: we need to add a require checking this is being called from the OGN contract.
```
require(msg.sender == address(tokenAddr), "Token must call");
```

Reply:
Thanks good catch added this